arcgis-ios-sdk-samples/Layers/Display WFS/DisplayWFSViewController.swift

```diff
@@ -0,0 +1,107 @@
+        wfsFeatureTable = AGSWFSFeatureTable(url: wfsServiceURL, tableName: wfsTableName)
+        
+        // Set feature request mode to manual - only manual is supported at v100.5.
+        // In this mode, you must manually populate the table - panning and zooming won't request features automatically.
+        wfsFeatureTable.featureRequestMode = .manualCache
+    
+        // Initialize a feature layer from WFS feature table
+        let wfsFeatureLayer = AGSFeatureLayer(featureTable: self.wfsFeatureTable)
+        
+        // Create a simple renderer
+        let simpleLineSymbol = AGSSimpleLineSymbol(style: .solid, color: .red, width: 3.0)
+        let renderer = AGSSimpleRenderer(symbol: simpleLineSymbol)
+        
+        // Apply renderer to feature layer
+        wfsFeatureLayer.renderer = renderer
+        
+        // Add feature layer to the map
+        map.operationalLayers.add(wfsFeatureLayer)
+        
+        // Query for features whenever map's viewpoint changes
+        self.mapView.viewpointChangedHandler = { [weak self] in
+            DispatchQueue.main.async {
+                self?.populateFeaturesFromQuery()
+            }
+        }
+        
+        // Set initial viewpoint
+        map.initialViewpoint = AGSViewpoint(targetExtent: AGSEnvelope(xMin: -122.341581, yMin: 47.613758, xMax: -122.332662, yMax: 47.617207, spatialReference: .wgs84()))
+        
+        // Assign the map to the map view
+        mapView.map = map
+
+        // Add the source code button item to the right of navigation bar
+        (navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["DisplayWFSViewController"]
+    }
+    
+    private func populateFeaturesFromQuery() {
+        // If there is an existing query request, cancel it
+        if let lastQuery = self.lastQuery{
+            lastQuery.cancel()
+        }
+        // Create query parameters
+        let params = AGSQueryParameters()
+        params.geometry = self.mapView.visibleArea?.extent
+        params.spatialRelationship = .intersects
+        
+        // Show progress
+        SVProgressHUD.show(withStatus: "Querying")
+    
+        // Populate features based on query
+        self.lastQuery = self.wfsFeatureTable.populateFromService(with: params, clearCache: true, outFields: ["*"]) { [weak self] (result: AGSFeatureQueryResult?, error: Error?) in
+            // Check and get results
+            if let result = result {
+                // The resulting features should be displayed on the map
+                // Print the count of features
+                print("Populated \(result.featureEnumerator().allObjects.count) features.")
+            }
+            // Check for error. If it's a user canceled error, do nothing.
+            // Otherwise, display an alert.
+            else if let error = error {
+                if error.localizedDescription.contains("User canceled error") {
```

Comment:
We normally check for this by testing the code of the error like so:

```swift
(error as NSError).code != NSUserCancelledError
```

Did you try that? If so, did it not work for you?

Reply:
That's a better way for sure. Will change it